### ISSUE TITLE
Ignoring media type for request body (Openapi 3)

### ISSUE THREAD

**Issue by jeremyschiemann** [schemathesis/schemathesis#472]
or at least they are not included in the case object.

Im converting case objects to curl commands for easier debugging and there is never any header present.

Im using this inside endpoint/post
```
"requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/simulationsRequest"
              }
            }
          }, 
          "description": "Body to give an input of desired simulations and their parameters."
        }
```

**Comment by jeremyschiemann** [comment]
about the curl commands, a CLI option definitely would be nice but since i use schemathesis in code i would like to see the curl in the case object to log it for example.

> it contains only data generated from an endpoint definition.
That is what I meant, I have an example for the requestbody in its schema, but if it gets tested shouldn't there be the content-type header in case.headers?

**Comment by Stranger6667** [comment]
Sorry for not answering you :(

> That is what I meant, I have an example for the requestbody in its schema, but if it gets tested shouldn't there be the content-type header in case.headers?

There should be a proper header, indeed. I'll check that

**Comment by Stranger6667** [comment]
The `Content-Type` header is included on the `requests` level - it is added automatically added if the `json` argument is passed to `requests.request`. And those keys in "content" are ignored, but I guess they should be included, it will also benefit cases where the schema specifies e.g. charset in this media type and the backend expects exactly the media type specified.

Extra:
- cURL issue - #689 
- tests generation for all media types - #690

**Comment by Stranger6667** [comment]
> Content-Type header specified as a parameter

It is forbidden [by the spec](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.3.md#fixed-fields-10):

> If in is "header" and the name field is "Accept", "Content-Type" or "Authorization", the parameter definition SHALL be ignored.

So, I guess, once #690 is finished, this issue can be closed as well

**Comment by Stranger6667** [comment]
Now all media types are taken into account when Schemathesis generates test data. Resolved via #862
